### ISSUE TITLE
NX 18: Vite config and plugins are called twice when a custom target is defined 

### ISSUE THREAD

**Issue by abarghoud** [nrwl/nx#22158]
### Current Behavior

When a target is defined in `project.json` using `@nx/vite:build` and 
```
"options": {
        "emptyOutDir": false,
        "outputPath": "dist/test",
        "configFile": "test/vite.config.ts"
      },
```
Running this target will execute plugins/vite.config.ts twice
Running the inferred `nx build` command works well.

This problem popped after I migrated from Nx 17 to Nx 18

### Expected Behavior

The vite.config.ts and plugins should only be called once.

### GitHub Repo

https://github.com/abarghoud/nx-vite-config-called-twice-bug

### Steps to Reproduce

1. use the repo
2. npm install
3. npm run build-from-custon-target (bugs occurs)

### Nx Report

```shell
Node   : 18.16.1
OS     : darwin-arm64
npm    : 9.5.1

nx                 : 18.0.7
@nx/js             : 18.0.7
@nx/linter         : 18.0.7
@nx/eslint         : 18.0.7
@nx/workspace      : 18.0.7
@nx/devkit         : 18.0.7
@nx/eslint-plugin  : 18.0.7
@nx/react          : 18.0.7
@nrwl/tao          : 18.0.7
@nx/vite           : 18.0.7
@nx/web            : 18.0.7
typescript         : 5.3.3
```

### Package Manager Version

yarn v1.22.19

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by abarghoud** [comment]
After some versions bisecting, the issue is reproducible starting from `@nx/vite@17.2.0` any version inferior to this works well.

**Comment by abarghoud** [comment]
@AgentEnder any idea on when this would be fixed ? Meanwhile, we rolled back to `@nx/vite@17.1.3` and `vite@4.5.2` temporarily to make it work.

**Comment by krivanek06** [comment]
@abarghoud , try version `18.0.8` . I had a similar problem, for it this version fixed it

**Comment by abarghoud** [comment]
@krivanek06 Still reproducible with `v18.0.8`.

**Comment by abarghoud** [comment]
@Coly010 Thank you for your response and for the clarification provided.

The double loading behavior is causing problems for me because it's directly impacting a custom Vite plugin that I've developed.
The plugin creates a WebSocketServer on init, so the double loading behavior causing the WebSocketServer to be initialized twice during the build process which leads the build to fail.

It's worth noting that in the older version, the Vite config was loaded only once, and as a result, I did not encounter any issues like this before the migration to Nx 18.
